File: summary/ruby/3_quick_sort.rb
def quick_sort(arr)
  return arr if arr.length <= 1

  pivot   = arr.sample
  less    = arr.select { |v| v < pivot }
  eq      = arr.select { |v| v == pivot }
  greater = arr.select { |v| v > pivot }

  quick_sort(less) + eq + quick_sort(greater)
end

arr = gets.chomp.split.map(&:to_i)
sorted_arr = quick_sort(arr)

puts "#{sorted_arr}"

File: summary/dart/3_quick_sort.dart
List<int> quickSort(List<int> arr) {
  if (arr.length <= 1) {
    return arr;
  }

  final pivot = arr[arr.length ~/ 2];
  final less = arr.where((v) => v < pivot).toList();
  final equal = arr.where((v) => v == pivot).toList();
  final greater = arr.where((v) => v > pivot).toList();

  return quickSort(less) + equal + quickSort(greater);
}

void main() {
  final arr = '10 7 8 9 1 5'.split(' ').map(int.parse).toList();
  final sortedArr = quickSort(arr);

  print(sortedArr);
}
